src/GraphQL.EntityFramework/PropertyAccess/PropertyCache.cs

```diff
@@ -31,16 +34,89 @@ public static Property<TInput> GetProperty(string path)
             });
     }
 
+    public static ParameterExpression GetSourceParameter()
+    {
+        if (sourceParam == null)
+        {
+            sourceParam = Expression.Parameter(typeof(TInput));
+        }
+
+        return sourceParam;
+    }
+
     static Expression AggregatePath(string path, Expression parameter)
     {
         try
         {
             return path.Split('.')
-                .Aggregate(parameter, Expression.PropertyOrField);
+                .Aggregate(parameter, (current, property) =>
+                    Expression.MakeMemberAccess(current, GetPropertyOrField(current.Type, property)));
         }
         catch (ArgumentException exception)
         {
             throw new Exception($"Failed to create a member expression. Type: {typeof(TInput).FullName}, Path: {path}. Error: {exception.Message}");
         }
     }
+
+    /// <summary>
+    /// Get Specified property or field member info of provided type
+    /// </summary>
+    /// <param name="type">Type to retrieve property from</param>
+    /// <param name="propertyOrFieldName">Name of property or field</param>
+    /// <returns></returns>
+    static MemberInfo GetPropertyOrField(Type type, string propertyOrFieldName)
```

Comment:
does this execute on every execution? shouldnt it cache the result?

Reply:
will look at doing that

Reply:
So that still uses your cache above in the entry function GetProperty above, the naming is a bit confusing considering the entry is "GetProperty" and this is "GetPropertyOrField", it's a copy and paste of the Microsoft "AggregatePath" function with some slight modifications to navigate the inheritance tree of interface types to find the requested property.

So for example getting the "Count" property of an IList type, you have to navigate to the ICollection type to get access to the "Count" property, just because of the way interface inheritance works in dotnet.

So the PropertyCache works the same as before except now it properly searches the inheritance tree for interface types for property/field info.